plugins/eos/gs-plugin-eos.c

```diff
@@ -1341,6 +1344,21 @@ gs_plugin_eos_blacklist_app_for_remote_if_needed (GsPlugin *plugin,
 	return do_blacklist;
 }
 
+static gboolean
+gs_plugin_eos_remove_blacklist_from_usb_if_needed (GsPlugin *plugin,
+						  GsApp *app)
+{
+	if (gs_app_has_category (app, "Blacklisted") &&
+	    gs_app_has_category (app, "USB")) {
+		g_debug ("Removing blacklisting from '%s': app is from USB", gs_app_get_unique_id (app));
+		gs_app_remove_category (app, "Blacklisted");
+
+		return TRUE;
+	}
```

Comment:
Nitpick: This is not too big a block, but we usually prefer early returns as it makes the code less hairy, especially if more logic is added in the future. (Remember to reomve the curly braces if you invert the returns)